corehq/toggles.py

```diff
@@ -127,22 +127,44 @@ def all_toggles():
     """
     Loads all toggles
     """
+    return all_toggles_by_name_in_scope(globals()).values()
+
+
+def all_toggles_by_name():
     # trick for listing the attributes of the current module.
     # http://stackoverflow.com/a/990450/8207
-    for toggle_name, toggle in globals().items():
+    return all_toggles_by_name_in_scope(globals())
+
+
+def all_toggles_by_name_in_scope(scope_dict):
+    result = {}
+    for toggle_name, toggle in scope_dict.items():
         if not toggle_name.startswith('__'):
             if isinstance(toggle, StaticToggle):
-                yield toggle
+                result[toggle_name] = toggle
+    return result
 
 
 def toggles_dict(username=None, domain=None):
     """
     Loads all toggles into a dictionary for use in JS
+
+    (only enabled toggles are included)
     """
     return {t.slug: True for t in all_toggles() if (t.enabled(username) or
                                                     t.enabled(domain))}
 
 
+def toggle_values_by_name(username=None, domain=None):
+    """
+    Loads all toggles into a dictionary for use in JS
+
+    all toggles (including those not enabled) are included
+    """
+    return {toggle_name: (toggle.enabled(username) or toggle.enabled(domain))
+            for toggle_name, toggle in all_toggles_by_name().items()}
```

Comment:
does this hit the database twice per toggle (once for `username` and once for `domain`)? If yes, would it make sense to refactor it to issue a single query that gets all toggles and their enabled/disabled values? (is that even possible?)